lib/Interpreter/CppInterOp.cpp

```diff
@@ -719,31 +719,46 @@ namespace Cpp {
     return ComputeBaseOffset(getSema().getASTContext(), DCXXRD, Paths.front());
   }
 
-  // FIXME: We should make the std::vector<TCppFunction_t> an out parameter to
-  // avoid copies.
-  std::vector<TCppFunction_t> GetClassMethods(TCppScope_t klass)
-  {
-
-    if (!klass)
-      return {};
-
+  void GetClassMethods(TCppScope_t klass, std::vector<TCppFunction_t> &methods) {
+    if (!klass) return;
+    
     auto *D = (clang::Decl *) klass;
 
     if (auto *TD = dyn_cast<TypedefNameDecl>(D))
       D = GetScopeFromType(TD->getUnderlyingType());
 
-    std::vector<TCppFunction_t> methods;
-    if (auto *CXXRD = dyn_cast_or_null<CXXRecordDecl>(D)) {
+    if (D && isa<CXXRecordDecl>(D)) {
```

Comment:
```suggestion
    if (auto* CXXRD = dyn_cast<CXXRecordDecl>(D)) {
```
`D` cannot be null here.

Reply:
D may not be null but we need the `isa<CXXRecordDecl(D)` check. otherwise this can cause tests to fail - `[with To = clang::CXXRecordDecl; From = clang::Decl`

Reply:
We can do something like this:

```cpp
if (!klass)
    return;

auto* D = (clang::Decl *) klass;

if (auto* TD = dyn_cast<TypedefNameDecl>(D))
  D = GetScopeFromType(TD->getUnderlyingType());

if (!D || !isa<CXXRecordDecl>(D))
  return;

auto* CXXRD = dyn_cast<CXXRecordDecl>(D);
```

Reply:
> D can be null if the cast to Decl fails. We also need the `isa<CXXRecordDecl(D)` check. Both these cases some tests to fail the cast- `[with To = clang::CXXRecordDecl; From = clang::Decl`

The cast to Decl cannot fail.

Reply:
Maybe its the attempt to get the scope from the underlying type, because if `(!isa<CXXRecordDecl>(D))` is just used instead of `if (!D || !isa<CXXRecordDecl>(D))`, I get this failure

 ```CppInterOp/llvm-project/llvm/include/llvm/Support/Casting.h:109: static bool llvm::isa_impl_cl<To, const From*>::doit(const From*) [with To = clang::CXXRecordDecl; From = clang::Decl]: Assertion `Val && "isa<> used on a null pointer"' failed.```

Reply:
Ok, that makes sense. Then maybe we will need an early exit there too.